ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/AlterTableHandler.java

```diff
@@ -168,10 +205,19 @@ public void handle(Context withinContext) throws Exception {
       }
     }
 
-    // If the tables are filtered based on name, then needs to handle the rename scenarios.
-    if (!withinContext.replScope.includeAllTables()) {
-      if (!handleForTableLevelReplication(withinContext)) {
-        LOG.info("Alter event for table " + before.getTableName() + " is skipped from dumping");
+    // Handle renames for table level replication.
+    if ((Scenario.RENAME == scenario)
+            && (!withinContext.replScope.includeAllTables() || withinContext.oldReplScope != null)) {
```

Comment:
These checks can be moved inside as follows. Will be less complex.
if (Scenario.RENAME == scenario) {
    ...
    boolean needDump = true;
    if (withinContext.oldReplScope != null) {
        needDump = handleRenameForReplacePolicy();
    } else if (!withinContext.replScope.includeAllTables()) {
        needDump = handleRenameForTableLevelReplication();
    }
    if (!needDump) {
        return;
    } 
}

Reply:
done